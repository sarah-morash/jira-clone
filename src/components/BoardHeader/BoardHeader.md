<!-- prettier- ignore -->
```
<BoardHeader title="Title" description="some great description here" />
```
