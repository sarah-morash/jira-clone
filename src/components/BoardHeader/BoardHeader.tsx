import * as React from "react";
import styled from "react-emotion";

interface IBoardHeader {
  /**
   * The name of the board
   */
  title: string;

  /**
   * Description of the board
   */
  description: string;
}

const BoardHeaderWrapper = styled.div`
  padding-bottom: 20px;
`;

const BoardTitle = styled.h1`
  margin: 0;
`;

const BoardDescription = styled.p`
  margin: 0;
  padding-top: 5px;
`;

export default (props: IBoardHeader) => (
  <BoardHeaderWrapper>
    <BoardTitle> {props.title} </BoardTitle>
    <BoardDescription> {props.description} </BoardDescription>
  </BoardHeaderWrapper>
);
