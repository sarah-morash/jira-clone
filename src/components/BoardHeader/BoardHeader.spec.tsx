import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import BoardHeader from "./BoardHeader";

describe("Board Header", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(
      <BoardHeader title="some name" description="some great description" />
    );
  });

  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  /**
   * Board Name
   */
  it("Should clearly render its name within an h1 tag", () => {
    wrapper.setProps({ title: "HELLO" });
    expect(wrapper.html()).toMatch("HELLO");
  });

  /**
   * Description
   */
  it("Should clearly display its description", () => {
    wrapper.setProps({ description: "some description" });
    expect(wrapper.html()).toMatch("some description");
  });
});
