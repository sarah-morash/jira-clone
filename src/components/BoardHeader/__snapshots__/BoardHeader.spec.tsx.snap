// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Header Should match its snapshot 1`] = `
<BoardHeaderWrapper>
  <BoardTitle>
     
    some name
     
  </BoardTitle>
  <BoardDescription>
     
    some great description
     
  </BoardDescription>
</BoardHeaderWrapper>
`;
